mantidproject/mantid
# Implement non-zero beam size in MonteCarloAbsorption algorithm

The `MonteCarloAbsortion` algorithm now has the concept of a beam size rather than all neutrons emanating from a zero-size point. The size of the beam is currently fixed to the width and height of the bounding box of the sample. This can be more flexible in the future.

**To test:**

The instructions for #15770 provide a script that can be used to drive the algorithm. The output no longer matches the integration algorithms as the average path lengths will be shorter as the beam is no longer divergent, resulting in slightly less absorption (higher transmitted factor)

Fixes #15887

Release Notes: I have decided not to update the release notes yet and wait until the general roadmap issue (#15443) is further along.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
